app/helpers/patient_query_helper.rb

```diff
@@ -472,8 +472,15 @@ def advanced_filter_relative_date(patients, field, filter, tz_diff, type)
       timespan = filter[:value][:number].to_i.months if filter[:value][:unit] == 'months'
       return patients if timespan.nil?
 
-      timeframe = { after: (timespan.ago - tz_diff).beginning_of_day, before: local_current_time } if filter[:value][:when] == 'past'
-      timeframe = { after: local_current_time, before: (timespan.from_now - tz_diff).end_of_day } if filter[:value][:when] == 'next'
+      case filter[:value][:operator]
+      when 'less-than'
+        timeframe = { after: timespan.ago - tz_diff, before: local_current_time } if filter[:value][:when] == 'past'
+        timeframe = { after: local_current_time, before: timespan.from_now - tz_diff } if filter[:value][:when] == 'future'
+      when 'more-than'
+        timespan += 1.day if filter[:filterOption][:hasTimestamp] == false
```

Comment:
Why exactly do we add a day if it's not a date with a timestamp?

Reply:
if you do more than 5 days ago for a relative date field without a timestamp, that includes the fifth day.  A decision was made by the PO team to not include that day so it is truly more than.

Decision was documented here: https://mitrecorp.slack.com/archives/C01A6P94R9B/p1613682229290700

Reply:
Ah okay, I remember that decision but this feels like an odd way to handle it (making it specific to the dates without timestamps). Technically, I think the results would also include the fifth day in your example if it is a date with a timestamp and the timestamp was at midnight (since it's just a >= comparison in the `patients_by_field_timeframe` method). I feel like the better way to handle this (and avoid that very specific edge case entirely) is to just have a separate method very similar to `patients_by_field_timeframe` that is only used for relative dates that just doesn't have `=` in the comparison. That does introduce some duplication, but I just feel like we're handling relative dates differently and as it is now it feels like a band-aid. Thoughts @timwongj @sgober?

Reply:
I'm kind of on the fence about that as both solutions don't seem too ideal. I wanted to consolidate the timeframe logic to make the code cleaner and concise but I also do agree that this is kind of a bandaid solution. I would at least add a comment explaining the purpose of this line of code

Reply:
Yeah agreed I don't love either. Also agreed at least a comment would be good.

Reply:
I will add a comment

Reply:
comment here: https://github.com/SaraAlert/SaraAlert/pull/703/commits/0f6b280029d11052d6ec1c1f6a09dbd86546bf2a